app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java

```diff
@@ -208,6 +213,21 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
         super.onRequestPermissionsResult(requestCode, permissions, grantResults);
     }
 
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        _isDPadPressed = isDPadKey(keyCode);
+        return super.onKeyDown(keyCode, event);
+    }
+
+    private boolean isDPadKey(int keyCode) {
+        return keyCode == KeyEvent.KEYCODE_DPAD_DOWN || keyCode == KeyEvent.KEYCODE_DPAD_UP || keyCode == KeyEvent.KEYCODE_DPAD_RIGHT || keyCode == KeyEvent.KEYCODE_DPAD_LEFT;
+    }
+
+    @Override
+    public void onTouch(View v, MotionEvent event) {
```

Comment:
Could you rename this to ``onEntryListTouch``? The name ``onTouch`` kind of sounds like we're listening for touch events for the whole activity.